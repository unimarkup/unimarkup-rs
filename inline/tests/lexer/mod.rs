use crate::Snapshot;
use unimarkup_commons::test_runner::{self, snap_test_runner::SnapTestRunner};
use unimarkup_inline::Tokenize;

mod snapshot;

#[test]
fn test_lexer_snapshots() {
    for case in crate::prepare_test_cases("spec/markup", "spec/snapshots/lexer") {
        let symbols = test_runner::scan_str(&case.input);
        let runner = SnapTestRunner::with_fn(&case.name, &symbols, |symbols| {
            let rest = &[];
            let snapshot = Snapshot::snap((case.input.as_ref(), symbols.tokens()));
            (snapshot, rest)
        })
        .with_info(format!("Test '{}' from '{}'", case.name, case.file_name));

        unimarkup_commons::run_snap_test!(runner, &case.out_path);
    }
}
