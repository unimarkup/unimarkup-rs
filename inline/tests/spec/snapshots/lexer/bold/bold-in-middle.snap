---
source: inline/tests/lexer/mod.rs
info: "Test 'bold-in-middle' from 'bold.yml'"
---
The next **word** is bold.
The next 
^^^^^^^^^ -> Plain @ (1:1)->(1:10)
         **
         ^^ -> Bold @ (1:10)->(1:12)
           word
           ^^^^ -> Plain @ (1:12)->(1:16)
               **
               ^^ -> Bold @ (1:16)->(1:18)
                  is bold.
                 ^^^^^^^^^ -> Plain @ (1:18)->(1:27)
                          ⏎
                          ^ -> Newline @ (1:27)->(2:1)

---
With input:

The next **word** is bold.


